argocd/schema_cluster.go

```diff
@@ -177,5 +177,29 @@ func clusterSchema() map[string]*schema.Schema {
 				},
 			},
 		},
+		"metadata": {
```

Comment:
can we also add `metadata.name` ? which I believe should allow to define predictable secret names, because by default argo will create secrets that are storing cluster configuration with weird names